src/skin/qml/qmlcontrolproxy.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <QtQml>
+#include <memory>
+
+#include "control/controlproxy.h"
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+class QmlControlProxy : public QObject {
+    Q_OBJECT
+// The REQUIRED flag only exists in Qt 5.14 and later.
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    Q_PROPERTY(QString group READ getGroup WRITE setGroup NOTIFY groupChanged REQUIRED)
+    Q_PROPERTY(QString key READ getKey WRITE setKey NOTIFY keyChanged REQUIRED)
+#else
+    Q_PROPERTY(QString group READ getGroup WRITE setGroup NOTIFY groupChanged)
+    Q_PROPERTY(QString key READ getKey WRITE setKey NOTIFY keyChanged)
+#endif
+    Q_PROPERTY(QString keyValid READ isKeyValid NOTIFY keyValidChanged)
+    Q_PROPERTY(QString initialized READ isInitialized NOTIFY initializedChanged)
+    Q_PROPERTY(double value READ getValue WRITE setValue NOTIFY valueChanged)
+    Q_PROPERTY(double parameter READ getParameter WRITE setParameter NOTIFY parameterChanged)
+
+  public:
+    QmlControlProxy(QObject* parent = nullptr);
+
+    void setGroup(const QString& group);
+    const QString& getGroup() const;
+
+    void setKey(const QString& key);
+    const QString& getKey() const;
+
+    void setValue(double newValue);
+    double getValue() const;
+
+    void setParameter(double newValue);
+    double getParameter() const;
+
+    bool isKeyValid() const;
+    bool isInitialized() const;
+
+  signals:
+    void groupChanged(const QString& group);
+    void keyChanged(const QString& key);
+    void keyValidChanged(bool valid);
```

Comment:
What's the purpose of the keyValidChanged() signal?

Reply:
Could be useful for skins: https://doc.qt.io/qt-5/qtqml-cppintegration-exposecppattributes.html#exposing-signals

But I just had an idea that we could make group and key immutable, which makes this unnecessary. Let's try this more restrictive approach. We can still loosen the restrictions if it turns out to be necessary. The reverse approach will be much harder.